osu.Game/Screens/Play/GameplayMenuOverlay.cs

```diff
@@ -194,14 +200,28 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
         {
         }
 
-        private void updateRetryCount()
-        {
-            // "You've retried 1,065 times in this session"
-            // "You've retried 1 time in this session"
+        [Resolved]
+        private IGameplayClock? gameplayClock { get; set; }
+
+        [Resolved]
+        private GameplayState? gameplayState { get; set; }
 
+        private void updateInfoText()
+        {
             playInfoText.Clear();
             playInfoText.AddText("Retry count: ");
             playInfoText.AddText(retries.ToString(), cp => cp.Font = cp.Font.With(weight: FontWeight.Bold));
+
+            if (gameplayState != null && gameplayClock != null)
+            {
+                (double firstHitTime, double lastHitTime) = gameplayState.Beatmap.CalculatePlayableBounds();
+
+                double progress = Math.Clamp((gameplayClock.CurrentTime - firstHitTime) / (lastHitTime - firstHitTime), 0, 1);
+
+                playInfoText.NewLine();
+                playInfoText.AddText("Song progress: ");
+                playInfoText.AddText(progress.ToString("0%"), cp => cp.Font = cp.Font.With(weight: FontWeight.Bold));
```

Comment:
At the risk of being the annoying stickler again, two issues here, one I know how to handle, and one I don't:

- If `firstHitTime == lastHitTime == gameplayClock.CurrentTime` - which is highly improbable, but not impossible - this will spit out `NaN`. Probably want to guard against this with a simple `IsNaN()` check and coercion to zero in such a case.
- The thing I don't know how to handle is that `SongProgressInfo`, from the triangles skin, calculates progress in a slightly different way:

  https://github.com/ppy/osu/blob/8a538a067571361a62c46977afbf589eee2f2d2b/osu.Game/Screens/Play/HUD/SongProgressInfo.cs#L147-L149

  Note the `Math.Floor()` in particular. It means that the progress on the pause screen and the progress on the progress bar can differ by 1%:

  ![image](https://github.com/ppy/osu/assets/20418176/6b072b03-e5e8-49c8-a457-501ae7815197)

  I'm honestly not sure which one is more correct. I think the flooring might be? Rationale for thinking that is we probably don't want to round 99.7% to 100%.

Reply:
Good catch on the `(int)` truncation. I think it makes more sense to have it, so I've added it at the new usage. Not sure about having a helper, as it won't meld well with the `SongProgressInfo` use case (which already localises the first/last hits).